src/main/java/uk/gov/hmcts/reform/iacaseaccessapi/infrastructure/service/CcdSupplementaryDetailsSearchService.java

```diff
@@ -28,18 +34,22 @@
     private final CoreCaseDataApi coreCaseDataApi;
     private final SystemTokenGenerator systemTokenGenerator;
     private final AuthTokenGenerator s2sAuthTokenGenerator;
+    private final int maxRecords;
+    private final List<HttpStatus> chunkedHttpStatusList = new ArrayList<>();
 
 
     public CcdSupplementaryDetailsSearchService(SystemTokenGenerator systemTokenGenerator,
                                                 CoreCaseDataApi coreCaseDataApi,
-                                                AuthTokenGenerator s2sAuthTokenGenerator) {
+                                                AuthTokenGenerator s2sAuthTokenGenerator,
+                                                @Value("${requestPagination.maxRecords}") int maxRecords) {
         this.systemTokenGenerator = systemTokenGenerator;
         this.coreCaseDataApi = coreCaseDataApi;
         this.s2sAuthTokenGenerator = s2sAuthTokenGenerator;
+        this.maxRecords = maxRecords;
     }
 
     @Override
-    public List<SupplementaryInfo> getSupplementaryDetails(List<String> ccdCaseNumberList) {
+    public List<SupplementaryInfo> getSupplementaryDetails(List<String> ccdCaseNumberList) throws ExecutionException, InterruptedException {
```

Comment:
you can keep it internally in the method (try - catch), I think throwing those two exceptions outside is not good